infrastructure/general/setup-full.md

```diff
@@ -0,0 +1,366 @@
+
+    ```bash
+    polkadot --name "Your Node's Name" --chain kusama
+    ```
+
+=== "macOS"
+
+    - Install Homebrew within the terminal by running:
+    ```bash
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+    ```
+    - Then, run:
+    ```bash
+    brew install openssl cmake llvm protobuf
+    ```
+
+    - Install Rust by running:
+    ```bash
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+    ```
+    - After Rust is installed, update and add the nightly version:
+    ```bash
+    # Ensure the current shell has cargo
+    source ~/.cargo/env
+
+    # Update the Rust toolchain
+    rustup default stable
+    rustup update
+
+    # Add the nightly and WebAssembly targets:
+    rustup update nightly
+    rustup target add wasm32-unknown-unknown --toolchain nightly
+    ```
+
+    - Verify your installation by running the following:
+    ```bash
+    rustup show
+
+    # You should see output similar to:
+
+    active toolchain
+    ----------------
+
+    stable-aarch64-apple-darwin (default)
+    rustc 1.68.1 (8460ca823 2023-03-20)
+    ```
+
+    ```bash
+    rustup +nightly show
+
+    # You should see output similar to:
+
+    installed targets for active toolchain
+    --------------------------------------
+
+    aarch64-apple-darwin
+    wasm32-unknown-unknown
+
+    active toolchain
+    ----------------
+
+    nightly-aarch64-apple-darwin (overridden by +toolchain on the command line)
+    rustc 1.71.0-nightly (9ecda8de8 2023-04-30)
+    ```
+
+    - Once Rust is configured, run the following command to clone and build the Polkadot code:
+    ```bash
+    git clone https://github.com/paritytech/polkadot-sdk polkadot-sdk
+    cd polkadot-sdk
+    ./scripts/init.sh
+    cargo build --release
+    ```
+    - Start your node:
+
+    ```bash
+    polkadot --name "Your Node's Name"
+    ```
+
+    - Find your node on [Telemetry](https://telemetry.polkadot.io/#list/Polkadot)
+
+=== "Windows"
+    - Install [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+    - Install [Ubuntu](https://docs.microsoft.com/en-us/windows/wsl/install-win10) (same webpage).
+    - Determine the latest version of the
+    [Polkadot binary](https://github.com/paritytech/polkadot-sdk/tree/master/polkadot/releases).
+    - Download the correct Polkadot binary within Ubuntu by running the following command. Replace
+    `*VERSION*` with the tag of the latest version from the last step (for example, `v0.8.22`):
+    ```bash
+    curl -sL https://github.com/paritytech/polkadot-sdk/releases/download/*VERSION*/polkadot -o polkadot
+    ```
+
+    - Then, run the following:
+    ```bash
+    sudo chmod +x polkadot
+    ```
+    - Start your node:
+    ```bash
+    ./polkadot --name "Your Node's Name"
+    ```
+
+    - Find your node on [Telemetry](https://telemetry.polkadot.io/#list/Polkadot){target=_blank}
```

Comment:
I'd add that you need to wait until your node is sync and then you will find it there